crates/ruff/src/directives.rs

```diff
@@ -107,15 +107,19 @@ pub fn extract_isort_directives(lxr: &[LexResult]) -> IsortDirectives {
         // omit a space after the colon. The remaining action comments are
         // required to include the space, and must appear on their own lines.
         let comment_text = comment_text.trim_end();
```

Comment:
It could help performance if you first test if the comment starts with either `# isort` or `ruff: isort` and, if not, bail out early. That ensures that we short-circuit for the majority of comments.